agent_plugin_builder/compare_package_lists.py

```diff
@@ -14,6 +14,6 @@ def take(n: int, iterable) -> list[str]:
     return list(islice(iterable, n))
 
 
-def all_equal(packages: set[str]) -> bool:
+def all_equal(packages: list[set[str]]) -> bool:
```

Comment:
`list` is too restrictive here. Use `Iterable`. Note that this is an [itertools recipe](https://docs.python.org/3/library/itertools.html#itertools-recipes). Since the groupby key is `None`, it should work for any type that can be compared via `==`, which AFAIK all objects have, so we could use `Iterable[Any]` here.